File: src/main/java/com/jss/osiris/modules/osiris/quotation/service/AssoServiceDocumentService.java
package com.jss.osiris.modules.osiris.quotation.service;

import com.jss.osiris.modules.osiris.quotation.model.AssoServiceDocument;

public interface AssoServiceDocumentService {
    public AssoServiceDocument getAssoServiceDocument(Integer id);

    public AssoServiceDocument addOrUpdateAssoServiceDocument(AssoServiceDocument AssoServiceDocument);
}

File: client-myjss/src/app/modules/my-account/services/asso.service.document.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { AppRestService } from '../../main/services/appRest.service';
import { AssoServiceDocument } from '../model/AssoServiceDocument';

@Injectable({
  providedIn: 'root'
})
export class AssoServiceDocumentService extends AppRestService<AssoServiceDocument> {


  constructor(http: HttpClient) {
    super(http, "quotation");
  }

  getAssoServiceDocument(assoServiceDocument: AssoServiceDocument): Observable<AssoServiceDocument> {
    return this.get(new HttpParams().set('idAssoServiceDocument', assoServiceDocument.id), "attachment/asso-service-document");
  }
}
